Marlin/src/module/planner.cpp

```diff
@@ -514,14 +514,23 @@ void Planner::check_axes_activity() {
         thermalManager.soft_pwm_amount_fan[2] = CALC_FAN_SPEED(2);
       #endif
     #else
-      #if HAS_FAN0
-        analogWrite(FAN_PIN, CALC_FAN_SPEED(0));
-      #endif
-      #if HAS_FAN1
-        analogWrite(FAN1_PIN, CALC_FAN_SPEED(1));
-      #endif
-      #if HAS_FAN2
-        analogWrite(FAN2_PIN, CALC_FAN_SPEED(2));
+      #ifndef ARDUINO  // reduce update rate to 10Hz rather than every time 
+                       //through idle loop (on LPC1768 was updating every 13uS)
+        static millis_t time_for_fan_update = 0;
+        if (millis() > time_for_fan_update) {
+          time_for_fan_update = millis() + 100; 
+      #endif    
+        #if HAS_FAN0
+          analogWrite(FAN_PIN, CALC_FAN_SPEED(0));
+        #endif
+        #if HAS_FAN1
+          analogWrite(FAN1_PIN, CALC_FAN_SPEED(1));
+        #endif
+        #if HAS_FAN2
+          analogWrite(FAN2_PIN, CALC_FAN_SPEED(2));
+        #endif
+      #ifndef ARDUINO
+        }
```

Comment:
I think this reduced timing should probably apply to the entire `check_axes_activity` function, since we don't need to compute any of these values so often. There might be other items in `idle()` that could also benefit from a reduced call frequency.